File: Task 1/C++/Components/Product/Tech.cpp
#include "Tech.h"

Tech::Tech() : Electronics(){
    type = "";
}

Tech::Tech(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string manufacturer, std::string type) 
            : Electronics(name, cost, supplier, discount_amount, packing_units, cost_per_unit, manufacturer){
    this->type = type;
};

Tech::Tech(const Tech& other) : Electronics(other) {
    this->type = other.type;
}

std::string Tech::display_product_info(){
    return Electronics::display_product_info()+
            "\nType: "+type;
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Tech.java
package com.logistics.Components.Product;

public class Tech extends Electronics {

    private String type;

    public Tech() {
        super();
        type = "";
    }

    public Tech(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String manufacturer, String type) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, manufacturer);
        this.type = type;
    }

    // Copy Constructor
    public Tech(Tech other) {
        super(other);
        this.type = other.type;
    }

    // Clone method
    @Override
    public Tech clone() {
        return new Tech(this);
    }

    // Getters
    public String getType() { return type; }

    // Setters
    public void setType(String type) { this.type = type; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nType: " + type;
    }
}
